src/lib/replication/log_store/repl_log_store.cpp

```diff
@@ -92,6 +92,11 @@ void ReplLogStore::end_of_append_batch(ulong start_lsn, ulong count) {
         for (auto const& rreq : *proposer_reqs) {
             if (rreq) { rreq->add_state(repl_req_state_t::LOG_FLUSHED); }
         }
+    } else {
+        // Ensure log flush even if reqs and proposer_reqs are empty, as this is the case for raft config entries.
+        RD_LOGT("Raft Channel: end_of_append_batch, no data to be written, only flush logs from {} , count {}",
+                start_lsn, count);
+        HomeRaftLogStore::end_of_append_batch(start_lsn, count);
```

Comment:
NIT, we can move this line out of judging area to make sure end_of_append_batch will be called for any case